### PR TITLE
alpine-make-rootfs: init

### PR DESCRIPTION
**Pull request by DanielSidhion**
## Description of changes

[alpine-make-rootfs](https://github.com/alpinelinux/alpine-make-rootfs) is a tool a bit similar to the already-packaged [alpine-make-vm-image](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/virtualization/alpine-make-vm-image/default.nix). Instead of creating a full vm image, it only creates an Alpine Linux rootfs.

I've been using this already on my own projects, thought I'd package it to make it easy for others to use it as well.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/virtualization/alpine-make-rootfs/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, lib, fetchFromGitHub, makeWrapper
+# All of the below are used in the alpine-make-rootfs script.
+, apk-tools, coreutils, findutils, gnugrep, gnused, gnutar, gzip, rsync, util-linux, wget
+}:
+stdenv.mkDerivation rec {
```
Comment (AndersonTorres) [inline pkgs/tools/virtualization/alpine-make-rootfs/default.nix:5]:
```suggestion
{ stdenv, lib, fetchFromGitHub, makeWrapper, apk-tools, coreutils, findutils, gnugrep, gnused, gnutar, gzip, rsync, util-linux, wget
}:

stdenv.mkDerivation rec {
```

File: pkgs/tools/virtualization/alpine-make-rootfs/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  pname = "alpine-make-rootfs";
+  version = "0.7.0";
+
+  src = fetchFromGitHub {
+    owner = "alpinelinux";
+    repo = "alpine-make-rootfs";
+    rev = "v${version}";
+    sha256 = "sha256-B5qYQ6ah4hFZfb3S5vwgevh7aEHI3YGLoA+IyipaDck=";
```
Comment (AndersonTorres) [inline pkgs/tools/virtualization/alpine-make-rootfs/default.nix:13]:
```suggestion
    hash = "sha256-B5qYQ6ah4hFZfb3S5vwgevh7aEHI3YGLoA+IyipaDck=";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/tools/virtualization/alpine-make-rootfs/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{ stdenv, lib, fetchFromGitHub, makeWrapper
```
Comment (AndersonTorres) [inline pkgs/tools/virtualization/alpine-make-rootfs/default.nix:1]: Is the GCC compiler a dependency somehow?

If not, use `stdenvNoCC` instead.

Comment (DanielSidhion) [reply pkgs/tools/virtualization/alpine-make-rootfs/default.nix]: Not a dependency, using `stdenvNoCC` now. Thanks for the review!

**Comment by nrabulinski** [comment]
Since it's a new package and it doesn't need any overrides, wouldn't it be better to put in in `by-name`?

**Review by nrabulinski — APPROVED** [review]

**Review by AndersonTorres — APPROVED** [review]
LGTM.

File: pkgs/by-name/al/alpine-make-rootfs/package.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  postInstall = ''
+    wrapProgram $out/bin/alpine-make-rootfs --set PATH ${lib.makeBinPath [
+      apk-tools coreutils findutils gnugrep gnused gnutar gzip rsync util-linux wget
+    ]}
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/alpinelinux/alpine-make-rootfs";
+    description = "Make customized Alpine Linux rootfs (base image) for containers";
+    maintainers = with maintainers; [ danielsidhion ];
```
Comment (eclairevoyant) [inline pkgs/by-name/al/alpine-make-rootfs/package.nix:28]:
```suggestion
    mainProgram = "alpine-make-rootfs";
    maintainers = with maintainers; [ danielsidhion ];
```

best to set `meta.mainProgram` when there's a main binary

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

**Review by eclairevoyant — APPROVED** [review]

**Pull request merged**